packages/compartment-mapper/src/node-modules.js

```diff
@@ -759,19 +844,101 @@ const graphPackages = async (
  * @param {Graph} graph
  * @param {Set<string>} conditions - build conditions about the target environment
  * for selecting relevant exports, e.g., "browser" or "node".
- * @param {SomePolicy} [policy]
- * @returns {CompartmentMapDescriptor}
+ * @param {TranslateGraphOptions} [options]
+ * @returns {PackageCompartmentMapDescriptor}
  */
 const translateGraph = (
   entryPackageLocation,
   entryModuleSpecifier,
   graph,
   conditions,
-  policy,
+  { policy, log = noop, packageDependenciesHook } = {},
 ) => {
-  /** @type {CompartmentMapDescriptor['compartments']} */
+  /** @type {Record<PackageCompartmentDescriptorName, PackageCompartmentDescriptor>} */
   const compartments = create(null);
 
+  /**
+   * Execute package dependencies hooks: default first (if policy exists), then user-provided.
+   *
+   * @param {CanonicalName} label
+   * @param {Record<string, FileUrlString>} dependencyLocations
+   * @returns {Record<string, FileUrlString>}
+   */
+  const executePackageDependenciesHook = (label, dependencyLocations) => {
+    const dependencies = new Set(
+      values(dependencyLocations).map(
+        dependencyLocation => graph[dependencyLocation].label,
+      ),
+    );
+
+    const packageDependenciesHookInput = {
+      canonicalName: label,
+      dependencies: new Set(dependencies),
+      log,
+    };
+
+    // Call default filter first if policy exists
+    let packageDependenciesHookResult;
+    if (policy) {
+      packageDependenciesHookResult = prePackageDependenciesFilter(
```

Comment:
(nit) nothing in prePackageDependenciesFilter name says it depends on policy. This if statement should be inside of it, but that's a total nitpick.